testing/fuzzing/fuzz_support.cc

```diff
@@ -4,18 +4,130 @@
-    return sys;
+    if (input.size < 2) {
+        return -1;
+    }
+
+    uint16_t fuzz_len = (input.data[0] << 8) | input.data[1];
+    input.data += 2;
+    input.size -= 2;
+
+    size_t available = input.size;
+
+    size_t res = fuzz_len > available ? available : fuzz_len;
+    res = buf_len > res ? res : buf_len;
+
+    memcpy(buf, input.data, res);
+    input.data += res;
+    input.size -= res;
+
+    return res;
 }
+
+static const Network_Funcs fuzz_network_funcs = {
+    .close = [](void *obj, int sock) { return 0; },
+    .accept = [](void *obj, int sock) { return 2; },
+    .bind = [](void *obj, int sock, const Network_Addr *addr) { return 0; },
+    .listen = [](void *obj, int sock, int backlog) { return 0; },
+    .recvbuf =
+        [](void *obj, int sock) {
+            // TODO(iphydf): Return something sensible here (from the fuzzer): number of
+            // bytes to be read from the socket.
+            return 0;
+        },
+    .recv =
+        [](void *obj, int sock, uint8_t *buf, size_t len) {
+            // Receive data from the fuzzer.
+            return recv_common(static_cast<Fuzz_System *>(obj)->data, buf, len);
+        },
+    .recvfrom =
+        [](void *obj, int sock, uint8_t *buf, size_t len, Network_Addr *addr) {
+            addr->addr = sockaddr_storage{};
+            // Dummy Addr
+            addr->addr.ss_family = AF_INET;
+
+            // We want an AF_INET address with dummy values
+            sockaddr_in *addr_in = reinterpret_cast<sockaddr_in *>(&addr->addr);
+            addr_in->sin_port = 12356;
+            addr_in->sin_addr.s_addr = INADDR_LOOPBACK + 1;
+            addr->size = sizeof(struct sockaddr);
+
+            return recv_common(static_cast<Fuzz_System *>(obj)->data, buf, len);
+        },
+    .send =
+        [](void *obj, int sock, const uint8_t *buf, size_t len) {
+            // Always succeed.
+            return static_cast<int>(len);
+        },
+    .sendto =
+        [](void *obj, int sock, const uint8_t *buf, size_t len, const Network_Addr *addr) {
+            // Always succeed.
+            return static_cast<int>(len);
+        },
+    .socket = [](void *obj, int domain, int type, int proto) { return 1; },
+    .socket_nonblock = [](void *obj, int sock, bool nonblock) { return 0; },
+    .getsockopt =
+        [](void *obj, int sock, int level, int optname, void *optval, size_t *optlen) {
+            memset(optval, 0, *optlen);
+            return 0;
+        },
+    .setsockopt = [](void *obj, int sock, int level, int optname, const void *optval,
+                      size_t optlen) { return 0; },
+};
+
+static const Random_Funcs fuzz_random_funcs = {
+    .random_bytes =
+        [](void *obj, uint8_t *bytes, size_t length) {
+            Fuzz_System *sys = static_cast<Fuzz_System *>(obj);
+            // Amount of data is limited
+            size_t available = sys->data.size;
+            size_t bytes_read = length > available ? available : length;
```

Comment:
good usecase for `std::min` if you want to clean that up

Reply:
Done.

Reply:
Done :). I did the one at the top of the file and missed this one.